### PR TITLE
Merge to main-test

### PR DESCRIPTION
**Pull request by Boschio**
Post/replies backend routes/models created

Event creation form styling added

Modified event page/registration form

Added new table to DB for storing API data

### CODE REVIEW

File: gameon-api/models/posts.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+class Posts {
+    static async createNewPost({post, user, eventId}) {
+        const requiredFields = ["postTitle", "postContent"]
+        requiredFields.forEach((field) => {
+            if (!post.hasOwnProperty(field) || !post[field]) {
+                throw new BadRequestError(`Required field - ${field} - missing from request body.`)
+            }
+        })
+        console.log("user: ", user)
```
Comment (LifeBringer) [inline gameon-api/models/posts.js:13]: Comment out

File: gameon-api/routes/events.js [hunk 2]
```diff
@@ -68,4 +70,74 @@ router.delete("/:eventId/withdraw", security.requireAuthenticatedUser, async (re
     }
 })
 
+router.post("/:eventId/posts", security.requireAuthenticatedUser, async (req, res, next) => {
+    try {
+        // create a new post for an event
+        const { user } = res.locals
+        console.log("user (routes): ", user)
```
Comment (LifeBringer) [inline gameon-api/routes/events.js:77]: Remove

File: gameon-api/routes/events.js [hunk 3]
```diff
@@ -68,4 +70,74 @@ router.delete("/:eventId/withdraw", security.requireAuthenticatedUser, async (re
...
+        const { eventId } = req.params
+        console.log("req.body: ", req.body)
```
Comment (LifeBringer) [inline gameon-api/routes/events.js:79]: Remove

File: gameon-api/routes/events.js [hunk 4]
```diff
@@ -68,4 +70,74 @@ router.delete("/:eventId/withdraw", security.requireAuthenticatedUser, async (re
...
+router.get("/:eventId/posts/:postId", async (req, res, next) => {
+    try {
+        // list a specified post for an event
+        console.log(req.params.eventId)
+        console.log(req.params.postId)
+        const eventId = req.params.eventId
+        console.log("eventId: ", eventId)
+        const postId = req.params.postId
+        console.log("postId: ", postId)
+
```
Comment (LifeBringer) [inline gameon-api/routes/events.js:107]: Remove

**Review by LifeBringer — CHANGES_REQUESTED** [review]
There are 36 console logs, please remove them prior to merge.

File: gameon-api/models/posts.js (hunk 1 above)
Comment (Boschio) [reply gameon-api/models/posts.js]: Removed most of these.

**Review by LifeBringer — APPROVED** [review]
LGTM

**Pull request merged**